src/arm/windows/init_by_logical_sys_info.c

```diff
@@ -0,0 +1,842 @@
+			(uint32_t) info_size);
+		return 0;
+	}
+	/* 3. Read the information structure */
+	if (GetLogicalProcessorInformationEx(info_type, infos, &info_size) == FALSE) {
+		cpuinfo_log_error("failed to query processor %"PRIu32" information: error %"PRIu32"",
+			(uint32_t)info_type, (uint32_t) GetLastError());
+		HeapFree(heap, 0, infos);
+		return 0;
+	}
+
+	/* 4. Parse the structure and store relevant data */
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info_end =
+		(PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX) ((uintptr_t) infos + info_size);
+	for (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info = infos;
+		info < info_end;
+		info = (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX) ((uintptr_t) info + info->Size))
+	{
+		if (info->Relationship != info_type) {
+			cpuinfo_log_warning(
+				"unexpected processor info type (%"PRIu32") for processor information",
+				(uint32_t) info->Relationship);
+			continue;
+		}
+
+		const uint32_t info_id = nr_of_structs++;
+		uint32_t group_count = 0;
+
+		switch(info_type) {
+			case RelationProcessorPackage:
+				parse_relation_processor_info(
+					processors,
+					number_of_processors,
+					global_proc_index_per_group,
+					info,
+					info_id,
+					cores,
+					chip_info);
+			break;
+			case RelationProcessorCore:
+				parse_relation_processor_info(
+					processors,
+					number_of_processors,
+					global_proc_index_per_group,
+					info,
+					info_id,
+					cores,
+					chip_info);
+			break;
+			case RelationCache:
+				parse_relation_cache_info(
+					processors,
+					number_of_processors,
+					caches,
+					numbers_of_caches,
+					global_proc_index_per_group,
+					info,
+					info_id);
+			break;
+			default:
+				cpuinfo_log_error(
+					"unexpected processor info type (%"PRIu32") for processor information",
+					(uint32_t) info->Relationship);
+			break;
+		}
+		
+	}
+	/* 5. Release dynamically allocated info structure. */
+	HeapFree(heap, 0, infos);
+	return nr_of_structs;
+}
+
+static void parse_relation_processor_info(
+	struct cpuinfo_processor* processors,
+	uint32_t nr_of_processors,
+	const uint32_t* global_proc_index_per_group,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info,
+	const uint32_t info_id,
+	struct cpuinfo_core* cores,
+	const struct woa_chip_info *chip_info)
+{
+	for (uint32_t i = 0; i < info->Processor.GroupCount; i++) {
+		const uint32_t group_id = info->Processor.GroupMask[i].Group;
+		/* Bitmask representing processors in this group belonging to this package */
+		KAFFINITY group_processors_mask = info->Processor.GroupMask[i].Mask;
+		while (group_processors_mask != 0) {
+			const uint32_t processor_id_in_group =
+				low_index_from_kaffinity(group_processors_mask);
+			const uint32_t processor_global_index =
+				global_proc_index_per_group[group_id] + processor_id_in_group;
+
+			if(processor_global_index >= nr_of_processors) {
+				cpuinfo_log_error("unexpected processor index %"PRIu32"",
+					processor_global_index);

```

Comment:
this shouldnt happen, but if it does then the store_package... calls are likely to throw